api.planx.uk/pay/pay.ts

```diff
@@ -0,0 +1,119 @@
+  res: Response,
+  next: NextFunction
+) {
+  // confirm that this local authority (aka team) has a pay token configured before creating the proxy
+  const isSupported =
+    process.env[`GOV_UK_PAY_TOKEN_${req.params.localAuthority.toUpperCase()}`];
+
+  const flowId = req.query?.flowId as string | undefined;
+  const sessionId = req.query?.sessionId as string | undefined;
+  const teamSlug = req.params.localAuthority;
+
+  if (isSupported) {
+    // drop req.params.localAuthority from the path when redirecting
+    // so redirects to plain [GOV_UK_PAY_URL] with correct bearer token
+    usePayProxy(
+      {
+        pathRewrite: (path) => path.replace(/^\/pay.*$/, ""),
+        selfHandleResponse: true,
+        onProxyRes: responseInterceptor(
+          async (responseBuffer, _proxyRes, _req, _res) => {
+            const responseString = responseBuffer.toString("utf8");
+            const govUkResponse = JSON.parse(responseString);
+            await logPaymentStatus({
+              sessionId,
+              flowId,
+              teamSlug,
+              govUkResponse,
+            });
+            return responseBuffer;
+          }
+        ),
+      },
+      req
+    )(req, res, next);
+  } else {
+    next({
+      status: 400,
+      message: `GOV.UK Pay is not enabled for this local authority`,
+    });
+  }
+}
+
+// exposed as /pay/:localAuthority/:paymentId and also used as middleware
+// fetches the status of the payment
+export const fetchPaymentViaProxy = fetchPaymentViaProxyWithCallback(
+  async (req: Request, govUkPayment: GovUKPayment) =>
+    postPaymentNotificationToSlack(req, govUkPayment)
+);
+
+export function fetchPaymentViaProxyWithCallback(
+  callback: (req: Request, govUkPayment: GovUKPayment) => Promise<void>
+) {
+  return async (req: Request, res: Response, next: NextFunction) => {
+    const flowId = req.query?.flowId as string | undefined;
+    const sessionId = req.query?.sessionId as string | undefined;
+    const teamSlug = req.params.localAuthority;
+
+    // will redirect to [GOV_UK_PAY_URL]/:paymentId with correct bearer token
+    usePayProxy(
+      {
+        pathRewrite: () => `/${req.params.paymentId}`,
+        selfHandleResponse: true,
+        onProxyRes: responseInterceptor(async (responseBuffer) => {
+          const govUkResponse = JSON.parse(responseBuffer.toString("utf8"));
+
+          await logPaymentStatus({
+            sessionId,
+            flowId,
+            teamSlug,
+            govUkResponse,
+          });
+
+          try {
+            await callback(req, govUkResponse);
+          } catch (e) {
+            next(e);
+            return "";
+          }
+
+          // only return payment status, filter out PII
+          return JSON.stringify({
+            payment_id: govUkResponse.payment_id,
+            amount: govUkResponse.amount,
+            state: govUkResponse.state,
+          });
+        }),
+      },
+      req
+    )(req, res, next);
+  };
+}
```

Comment:
This function wraps the payment fetch proxy and inserts a callback. This allows fetching payments to follow the same pattern (call logPaymentStatus + return filtered body) while making the function reusable for invite-to-pay.